src/Reduce.php

```diff
@@ -175,4 +175,32 @@ public static function isReversed(iterable $data): bool
 
         return true;
     }
+
+    /**
+     * Returns true if given collection is exactly the same as another ones.
+     *
+     * @param iterable<mixed> $data
+     * @param iterable<mixed> $n
+     * @param iterable<mixed> ...$extra
+     *
+     * @return bool
+     */
+    public static function exactlyN(iterable $data, iterable $n, iterable ...$extra): bool
```

Comment:
Suggestion: How about naming this `same` or something. `exactlyN` generally has a different meaning, as in, the iterable/collection has exactly n elements (for example, exactly 10 elements). It seems like the function you are proposing is "do the iterables produce the same collection of elements?" Therefore, `same` seems to be a better fit here.

Reply:
Done.